ImageTaggingApp.Console.App/Commands/TagCommand.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections.Generic;
+using ImageTaggingApp.Console.App.Entities;
+using ImageTaggingApp.Console.App.Factories;
+
+namespace ImageTaggingApp.Console.App.Commands {
+    public class TagCommand : ICommand {
+        private readonly IImageProcessingFactory _imageProcessingFactory;
+        private readonly string _imagesPath;
+
+        public TagCommand(IImageProcessingFactory imageProcessingFactory, string imagesPath) {
+            _imageProcessingFactory = imageProcessingFactory;
+            _imagesPath = imagesPath;
+        }
+
+        public void Execute() {
+            IReadOnlyCollection<string> imageFilePaths = _imageProcessingFactory.SearchImageFilesToBeTagged(_imagesPath);
```

Comment:
Took a look at pipeline pattern. Great suggestion for using it. It added more flexibility and solved the problem with temporary results caching. The pipeline I have made has 3 main stages: preprocess (at the moment identify blurry images), tagging and saving results to db. These 3 stages works concurrently. Therefore, any processed and tagged image is saved to db automatically.